ai_genomics/analysis/gtr_definitions.py

```diff
@@ -0,0 +1,130 @@
+
+    # Topic distribution
+    topic_distr = pd.Series(Counter([topic["text"] for topic in topics]))
+
+    relevant_concepts = topic_distr.loc[
+        [
+            any(t in conc.lower() for t in config["gtr_topic_search"])
+            for conc in topic_distr.index
+        ]
+    ]
+
+    logging.info(relevant_concepts)
+
+    # Get the project with relevant topics from the config file
+    ai_topics, genomics_topics = [
+        config[f"gtr_{disc}_concepts"] for disc in ["ai", "genom"]
+    ]
+
+    ai_projs, genom_projs = [
+        set(
+            [
+                element["project_id"]
+                for element in topics
+                if any(t in element["text"] for t in terms)
+            ]
+        )
+        for terms in [ai_topics, genomics_topics]
+    ]
+
+    logging.info(f"AI projects: {len(ai_projs)}")
+    logging.info(f"Genomics projects: {len(genom_projs)}")
+
+    # Look at abstracts
+    ai_abstract_terms, genom_abstract_terms = [
+        config[f"gtr_{t}_abstract"] for t in ["ai", "genom"]
+    ]
+
+    ai_projs_abstr, genom_projs_abstr = [
+        set(
+            [
+                element["id"]
+                for element in projects
+                if any(  # If any of the abstract terms are in the abstract or title
+                    t
+                    in str(element["abstractText"]).lower()
+                    + str(element["title"]).lower()
+                    for t in terms
+                )
+            ]
+        )
+        for terms in [ai_abstract_terms, genom_abstract_terms]
+    ]
+
+    logging.info(f"Projects with AI abstract / title: {len(ai_projs_abstr)}")
+    logging.info(f"Projects with genomics abstract / title: {len(genom_projs_abstr)}")
+    logging.info(
+        f"Projects with AI and genomics abstract / title: {len(ai_projs_abstr & genom_projs_abstr)}"
+    )
+
+    #  Intersection of the two sets
+    unified_projects = set(
+        ai_projs_abstr.union(ai_topics) & genom_projs_abstr.union(genomics_topics)
+    )
+
+    # Extract projects (NB there are duplicates in the projects list)
+    rel_projects = [proj for proj in projects if proj["id"] in unified_projects]
+
+    ai_genomics_combined = []
+    used_ids = []
+
+    # We use this to catch dupes
+    for _id in set([proj["id"] for proj in rel_projects]):
+        for p in rel_projects:
+            if _id in used_ids:
+                pass
+            else:
+                if p["id"] == _id:
+                    ai_genomics_combined.append(p)
+                    used_ids.append(_id)
+
+    logging.info(f"AI and genomics projects combined: {len(ai_genomics_combined)}")
+
+    # Get project examples
+    project_examples = []
+
+    for sampled in random.sample(ai_genomics_combined, 5):
+
+        project_examples.append(
+            {
+                "Project title": sampled["title"],
+                "Abstract (truncated)": sampled["abstractText"][:700] + "...",
+            }
+        )
```

Comment:
Can use a list comp here

```suggestion
    # Get project examples
    project_examples = [
        {
            "Project title": sampled["title"],
            "Abstract (truncated)": sampled["abstractText"][:700] + "...",
        }
        for sampled in random.sample(ai_genomics_combined, 5)
    ]
```